nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -4506,6 +4515,123 @@ public ComponentHistoryDTO getComponentHistory(final String componentId) {
         return history;
     }
 
+    private ControllerServiceEntity createControllerServiceEntity(final String serviceId, final NiFiUser user) {
+        final ControllerServiceNode serviceNode = controllerServiceDAO.getControllerService(serviceId);
+        return createControllerServiceEntity(serviceNode, Collections.emptySet(), user);
+    }
+
+    @Override
+    public ProcessorDiagnosticsEntity getProcessorDiagnostics(final String id) {
+        final ProcessorNode processor = processorDAO.getProcessor(id);
+        final ProcessorStatus processorStatus = controllerFacade.getProcessorStatus(id);
+
+        // Generate Processor Diagnostics
+        final NiFiUser user = NiFiUserUtils.getNiFiUser();
+        final ProcessorDiagnosticsDTO dto = controllerFacade.getProcessorDiagnostics(processor, processorStatus, bulletinRepository, serviceId -> createControllerServiceEntity(serviceId, user));
+
+        // Filter anything out of diagnostics that the user is not authorized to see.
+        final List<JVMDiagnosticsSnapshotDTO> jvmDiagnosticsSnaphots = new ArrayList<>();
+        final JVMDiagnosticsDTO jvmDiagnostics = dto.getJvmDiagnostics();
+        jvmDiagnosticsSnaphots.add(jvmDiagnostics.getAggregateSnapshot());
+
+        // filter controller-related information
+        final boolean canReadController = authorizableLookup.getController().isAuthorized(authorizer, RequestAction.READ, user);
+        if (!canReadController) {
+            for (final JVMDiagnosticsSnapshotDTO snapshot : jvmDiagnosticsSnaphots) {
+                snapshot.setMaxEventDrivenThreads(null);
+                snapshot.setMaxTimerDrivenThreads(null);
+                snapshot.setBundlesLoaded(null);
+            }
+        }
+
+        // filter system diagnostics information
+        final boolean canReadSystem = authorizableLookup.getSystem().isAuthorized(authorizer, RequestAction.READ, user);
+        if (!canReadSystem) {
+            for (final JVMDiagnosticsSnapshotDTO snapshot : jvmDiagnosticsSnaphots) {
```

Comment:
What about open and max file descriptors?

Reply:
That's a good call. Will update that.